hooks/pre-commit

```diff
@@ -2,8 +2,9 @@
 
 set -eu
 
-golint
+golint ./...
 go fmt
+go tool vet .
 go tool vet --shadow .
-gocyclo -over 12 .
-go test
+gocyclo -over 19 .
+go test .
```

Comment:
The `.` shouldn't be necessary (works fine without on `gomatrix` which uses the same directory structure).

Also, you probably want to make this:
```
go test -timeout 5s -test.v
```

which sets a 5s timeout for each test and logs test names and stdout which can be useful when debugging problems